frontend/src/SettingsDashboard.jsx

```diff
@@ -145,20 +160,32 @@ const SettingsTable = ({ name, settings, settingsType, onEditCell }) => {
         },
     ];
 
+    let settingsModal = null;
+    if (modalOpen) {
+        settingsModal = <SettingsModal
+            type={settingsType}
+            onClose={handleModalClose}
+            action={modalAction}
+            setting={modalData.setting}
+            value={modalData.value}
+            open={true}
+        />;
+    }
+
     return <Grid container direction="column" alignItems="flex-start" justify="flex-start" style={{height: '100%'}}>
         <Grid container direction="row" alignItems="center" justify="space-between">
             <Grid item>
                 <Typography variant="h5">{name}</Typography>
             </Grid>
             <Grid item>
-                <IconButton onClick={handleOpenModal}>
+                <IconButton onClick={() => handleOpenModal()}>
                     <AddBoxIcon />
                 </IconButton>
-                <NewSettingsModal type={settingsType} onClose={handleModalClose} open={modalOpen} />
+                {settingsModal}
             </Grid>
         </Grid>
         <Grid item style={{height: 'calc(100% - 48px)', width: '100%'}}>
-            {!!settings.length && <Table config={tableConfig} data={settings} onCellEdit={handleCellEdit} />}
+            {!!settings.length && <Table config={tableConfig} data={settings} />}
```

Comment:
I think this also means that we should remove `editable: true` from `tableConfig` for both columns. And since this is the only place that uses it, I wouldn't be against removing the entire editable cell logic from `Table`